node/src/main/scala/com/wavesplatform/state/Blockchain.scala

```diff
@@ -108,6 +108,8 @@ trait Blockchain {
   // the following methods are used exclusively by patches
   def allActiveLeases: CloseableIterator[LeaseTransaction]
 
+  def proofForBalanceOnHeight(address: Address, height: Height): Option[ProvenBalance]
```

Comment:
To make naming consistent with block (header) at height and other height-related methods:
```suggestion
  def balanceProof(address: Address, height: Height): Option[ProvenBalance]
```